### PR TITLE
List and Search: Kinda everything

### PR DESCRIPTION
**Pull request by suriruhani**
currently excludes system tests and doc diagrams, to be included in the following PR today

- includes new addition to search logic : autocorrection/typo management
- includes new addition to list logic : takes two optional arguments
- docs updated
- tests fixed

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -219,17 +219,26 @@ Edits the 3rd source, changing the type and author of the source to be `Journal`
 * `edit 5 t/` +
 Edits the 5th source and clears all its existing tags.
 
+// tag::search[]
```
Comment (case141) [inline docs/UserGuide.adoc:222]: I don't think you need a tag in the UserGuide, travis will report an asciidocs error.

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -247,26 +258,67 @@ Examples:
...
-Returns any source having tags `ML` and having the word `intelligence` somewhere in their content
+* `search i/wiki y/web d/intelligence t/ML` +
+Returns any source(s) having tags `ML` and having the word `intelligence` somewhere in their content
 (detail) and having a type of `website` or `web series`.
+* `search i/data i/algo` +
+Returns the source(s) with both substrings 'data' and 'algo' included in them.
+* `search y/wesbite`
+Returns the source(s) of type 'website' [and other similar strings, if any]
+// end::search[]
+// tag::list[]
```
Comment (case141) [inline docs/UserGuide.adoc:269]: same here

File: docs/UserGuide.adoc (hunk 1 above)
Comment (suriruhani) [reply docs/UserGuide.adoc:222]: No, on the contrary, the error was when we included this tag in our PPP but did *not* tag in the UG. This is needed to include UG contributions in PPP. I check, no error or warning.
@DarrenDragonLee requested to do the same.

Comment (case141) [reply docs/UserGuide.adoc:222]: alright then

**Review by case141 — APPROVED** [review]

**Pull request merged**